app/src/test/java/org/simple/clinic/analytics/AnalyticsTest.kt

```diff
@@ -147,14 +152,32 @@ class AnalyticsTest {
     )
   }
 
+  @Test
+  fun `when a reporter fails clearing the user id, no  error should be thrown`() {
+    Analytics.addReporter(FailingAnalyticsReporter())
+    Analytics.clearUserId()
+  }
+
   @Test
   fun `when setting the user id, the property must also be set on the reporters`() {
     val reporter = MockAnalyticsReporter()
     Analytics.addReporter(reporter)
 
     val uuid = UUID.randomUUID()
     Analytics.setUserId(uuid)
-    assertThat(reporter.setProperties).isEqualTo(mapOf("userId" to uuid.toString()))
+    assertThat(reporter.setUserId).isEqualTo(uuid.toString())
+  }
+
+  @Test
+  fun `when clearing the user id, the user ID must be cleared from the reporters`() {
+    val reporter = MockAnalyticsReporter()
+    Analytics.addReporter(reporter)
+
+    val uuid = UUID.randomUUID()
+    Analytics.setUserId(uuid)
+    assertThat(reporter.setUserId).isEqualTo(uuid.toString())
```

Comment:
`setUserId` is a property. Can this be renamed to just `userId`?

Reply:
My intention was to capture the meaning `the user ID that was set on this reporter`. I guess `userId` will work just as well and won't clash with the convention of the Java setter method.